pkg/trace/api/api_test.go

```diff
@@ -586,6 +588,253 @@ func TestHandleStats(t *testing.T) {
 	})
 }
 
+// TestInfoHandler ensures that the keys returned by the /info handler do not
+// change from one release to another to ensure consistency. Tracing clients
+// depend on these keys to be the same. The chances of them changing are quite
+// high if anyone ever modifies a field name in the (*AgentConfig).Config structure.
+//
+// * In case a field name gets modified, the `json:""` struct field tag
+// should be used to ensure the old key is marshalled for this endpoint.
+func TestInfoHandler(t *testing.T) {
+	u, err := url.Parse("http://localhost:8888/proxy")
+	if err != nil {
+		log.Fatal(err)
+	}
+	jsonObfCfg := config.JSONObfuscationConfig{
+		Enabled:            true,
+		KeepValues:         []string{"a", "b", "c"},
+		ObfuscateSQLValues: []string{"x", "y"},
+	}
+	obfCfg := &config.ObfuscationConfig{
+		ES:                   jsonObfCfg,
+		Mongo:                jsonObfCfg,
+		SQLExecPlan:          jsonObfCfg,
+		SQLExecPlanNormalize: jsonObfCfg,
+		HTTP: config.HTTPObfuscationConfig{
+			RemoveQueryString: true,
+			RemovePathDigits:  true,
+		},
+		RemoveStackTraces: false,
+		Redis:             config.Enablable{Enabled: true},
+		Memcached:         config.Enablable{Enabled: false},
+	}
+	rcv := newTestReceiverFromConfig(&config.AgentConfig{
+		Enabled:    true,
```

Comment:
I cannot tell from the test.  How are configuration variables that are not set handled?
I'd like to always receive a value even if its the default.

Reply:
You can take a look here: https://github.com/DataDog/datadog-agent/blob/fcc68a02c6c146826a302db9b39df3c761f7aac3/pkg/trace/config/config.go#L50-L118.

In that structure, all values will show up regardless of their value, except the ones marked with `json:",omitempty"`. I think it's really weird for those to show up as `null`. But, if you feel strongly, I can have them like that.